### ISSUE TITLE
Invalid server code generated for file response

### ISSUE THREAD

**Issue by onokonem** [go-swagger/go-swagger#1195]
## Problem statement

Code generated by go-swagger could not be compiled

## Swagger specification

```
swagger: "2.0"
info:
  description: API + File download
  title: Test
  version: 1.0.0
schemes:
- http
consumes:
- application/json
produces:
- application/json
paths:
  /some/file.txt:
    get:
      operationId: someFile
      produces:
      - application/octet-stream
      responses:
        200:
          description: download file
          schema:
            type: file
```

## Steps to reproduce

```
swagger generate server --target=gen

go build ./gen/cmd-server
```

Expecting: build success

Got:

```
gen/models/some_file_o_k_body.go:20:6: invalid receiver type SomeFileOKBody (SomeFileOKBody is a pointer type)
```

## Environment
swagger version: dev  
go version: go version go1.9 darwin/amd64  
OS:   OSX 10.11.6

**Comment by casualjim** [comment]
the type file is specific for formData params.
I believe your intent is to stream some data to the client, perhaps with the content-disposition header. To do this use:

```yaml
type: string
format: binary
```

This will render the correct code for you to use.

**Comment by casualjim** [comment]
this is fixed on master by: #1329
